fillers/eips/eip3860.py

```diff
@@ -233,504 +259,394 @@ def generate_tx_initcode_limit_test_cases(
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=SINGLE_BYTE_INITCODE,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_32_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_33_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_49120_BYTES,
-        eip_3860_active=True,
-    )
-
-    yield from generate_gas_cost_test_cases(
-        initcode=INITCODE_ZEROS_49121_BYTES,
-        eip_3860_active=True,
-    )
-
 
 """
 Test cases using the CREATE opcode
 """
 
 
-def generate_create_opcode_initcode_test_cases(
-    opcode: str,
-    initcode: Initcode,
-    eip_3860_active: bool,
-):
+@pytest.mark.parametrize(
+    "initcode",
+    [
+        INITCODE_ZEROS_MAX_LIMIT,
+        INITCODE_ONES_MAX_LIMIT,
+        INITCODE_ZEROS_OVER_LIMIT,
+        INITCODE_ONES_OVER_LIMIT,
+        EMPTY_INITCODE,
+        SINGLE_BYTE_INITCODE,
+        INITCODE_ZEROS_32_BYTES,
+        INITCODE_ZEROS_33_BYTES,
+        INITCODE_ZEROS_49120_BYTES,
+        INITCODE_ZEROS_49121_BYTES,
+    ],
+    ids=get_initcode_name,
+)
+@pytest.mark.parametrize("opcode", ["create", "create2"], ids=lambda x: x)
+class TestCreateInitcode:
     """
-    Generates a StateTest using the `CREATE`/`CREATE2` opcode based on the
-    provided `initcode`, its executing cost, and the deployed code.
+    Test contract creation via the CREATE/CREATE2 opcodes that have an initcode
+    that is on/over the max allowed limit.
     """
-    env = Environment()
-
-    call_code = Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE)
-    call_code += Op.SSTORE(
-        Op.CALL(5000000, 0x100, 0, 0, Op.CALLDATASIZE, 0, 0),
-        1,
-    )
 
-    create_code = Template(
-        """
-        {
-            let contract_length := calldatasize()
-            calldatacopy(0, 0, contract_length)
-            let gas1 := gas()
-            let res := ${create_instruction}
-            let gas2 := gas()
-            sstore(0, res)
-            sstore(1, sub(gas1, gas2))
-        }
-        """
-    )
-
-    if opcode == "create":
-        code = Yul(
-            create_code.substitute(
-                create_instruction="create(0, 0, contract_length)"
-            )
-        )
-        created_contract_address = compute_create_address(
-            address=0x100,
-            nonce=1,
+    @pytest.fixture
+    def create_code(self, opcode: str):  # noqa: D102
+        create_code = Template(
+            """
+            {
+                let contract_length := calldatasize()
+                calldatacopy(0, 0, contract_length)
+                let gas1 := gas()
+                let res := ${create_instruction}
+                let gas2 := gas()
+                sstore(0, res)
+                sstore(1, sub(gas1, gas2))
+            }
+            """
```

Comment:
```suggestion
        create_code = (
            Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE)
            + Op.GAS
            + create_call
            + Op.GAS
            # stack: [Gas 2, Call Result, Gas 1]
            + Op.SWAP1
            # stack: [Call Result, Gas 2, Gas 1]
            + Op.SSTORE(0)
            # stack: [Gas 2, Gas 1]
            + Op.SWAP1
            # stack: [Gas 1, Gas 2]
            + Op.SUB
            # stack: [Gas 1 - Gas 2]
            + Op.SSTORE(1)
        )
```
We can rewrite as such, where `create_call` depends on the `opcode` parameter. The only other change required is on line 580 with the gas calculation would be a bit affected by this.

Reply:
Refactored in ec3490c07f25ad00845d6b5a27ed7f886bc5631e. Could you please check the `create_call` and gas calculation logic?